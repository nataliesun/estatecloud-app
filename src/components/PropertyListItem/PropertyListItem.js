import React, { Component } from 'react';

import './PropertyListItem.scss';

class PropertyListItem extends Component {



  render() {
    const { address, status, dates, rent_price, mortgage_payment } = this.props;
    return (
      <li className="PropertyListItem">
        <h2>{capitalizeAddress(address)}</h2>
        <p>{checkStatus(status, dates)}</p>
        {
          !!Number(rent_price)
          && <p>
            Rented at: ${rent_price}
          </p>
        }
        <p>Monthly profit: ${calculateProfit(mortgage_payment, rent_price)}</p>
        <button className="edit-btn">Edit</button>
        <button className="delete-btn">Delete</button>
      </li>
    );
  }
}

export default PropertyListItem;

function capitalizeAddress(address) {
  const splitAddress = address.toLowerCase().split(' ');

  for (let i = 0; i < splitAddress.length; i++) {
    splitAddress[i] = splitAddress[i].charAt(0).toUpperCase() + splitAddress[i].substring(1);
  }

  return splitAddress.join(' ');
}

function checkStatus(status, dates) {
  if (status === 'occupied') {
    return `Occupied until: ${dates[1].toLocaleDateString()}`
  } else if (status === 'rented') {
    return `Rented until: ${dates[1].toLocaleDateString()}`
  } else {
    return `Available`
  }
}

function calculateProfit(mortgage, rent) {
  return Number(rent) - Number(mortgage)
}