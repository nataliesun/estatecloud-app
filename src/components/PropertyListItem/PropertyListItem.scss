@import '../variables.scss';

.PropertyListItem {
  list-style-type: none;
  // border-bottom: 1px solid rgb(218, 218, 218);
  min-height: 20vh;
  box-shadow:   0 4px 8px 0 rgba(172, 172, 172, 0.2), 0 6px 20px 0 rgba(145, 145, 145, 0.19);
  margin-bottom: 20px;
  border-radius: 5px;

  &:first-of-type {
    // border-top: 1px solid rgb(218, 218, 218);
  }

  h2 {
    padding: 20px 0;
  }

  .edit-btn {
    @include highlighted-button($button-highlight);
    margin-top: 10px;
    margin-bottom: 10px;
    margin-right: 10px;
    cursor: pointer;
  }

  .delete-btn {
    @include highlighted-button($button-purple);
    cursor: pointer;
  }

  a {
    text-decoration: none;
    color: $main-text-color;
  }
}