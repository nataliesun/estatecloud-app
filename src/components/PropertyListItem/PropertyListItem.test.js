import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import PropertyListItem from './PropertyListItem';

describe(`PropertyListItem Component`, () => {

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <PropertyListItem />
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <PropertyListItem />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})