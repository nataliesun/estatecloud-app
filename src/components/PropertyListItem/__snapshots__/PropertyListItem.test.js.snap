// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyListItem Component renders the UI as expected 1`] = `
<li
  className="PropertyListItem"
>
  <a
    href="/property/undefined"
    onClick={[Function]}
  >
    <h2 />
  </a>
  <p>
    Available until: tbd
  </p>
  <p>
    Monthly profit: $
  </p>
  <button
    className="edit-btn"
    onClick={[Function]}
  >
    Edit
  </button>
  <button
    className="delete-btn"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
